### PR TITLE
Print base rings terse and starting with lowercase

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

**Comment by fingolfin** [comment]
There are still some uses of `compact` left but they all seem to be related to printing vectors or elements of a matrix, so I am not sure if they should be kept at `compact` or changed to supercompact ` terse ... E.g.:
```julia
function show(io::IO, v::SNFModuleElem)
   print(io, "(")
   len = ngens(parent(v))
   for i = 1:len - 1
      print(IOContext(io, :compact => true), v.v[1, i])
      print(io, ", ")
   end
   if len > 0
      print(IOContext(io, :compact => true), v.v[1, len])
   end
   print(io, ")")
end
```

**Comment by lgoettgens** [comment]
The failure https://github.com/Nemocas/AbstractAlgebra.jl/actions/runs/8816772846/job/24201674662?pr=1680#step:7:3437 looks unexpected.
```
Parent object call overloading: Test Failed at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/oscar-dev/Nemo/test/HeckeMiscLocalization-test.jl:78
  Expression: L1(L2(5))
    Expected: ErrorException
      Thrown: MethodError
Stacktrace:
 [1] macro expansion
   @ ~/work/AbstractAlgebra.jl/AbstractAlgebra.jl/oscar-dev/Nemo/test/HeckeMiscLocalization-test.jl:78 [inlined]
```
I'll look into this

**Comment by lgoettgens** [comment]
> There are still some uses of `compact` left but they all seem to be related to printing vectors or elements of a matrix, so I am not sure if they should be kept at `compact` or changed to supercompact ` terse ... E.g.:

Most of them seem to be AA `MatElem`s. But element types don't have the three types of printing anyway. So no idea...

File: src/generic/Misc/Localization.jl [hunk 1]
```diff
@@ -167,9 +167,9 @@ end
 
 function show(io::IO, L::LocalizedEuclideanRing)
    if L.comp
```
Comment (lgoettgens) [inline src/generic/Misc/Localization.jl:169]:
```suggestion
   io = pretty(io)
   if L.comp
```

File: src/generic/FunctionField.jl [hunk 2]
```diff
@@ -772,7 +772,7 @@ end
 @enable_all_show_via_expressify FunctionFieldElem
 
 function show(io::IO, R::FunctionField)
-   print(IOContext(io, :compact => true), "Function Field over ",
+   print(terse(io), "Function Field over ",
```
Comment (lgoettgens) [inline src/generic/FunctionField.jl:775]: Lowercase missing for base_ring

File: src/generic/QuotientModule.jl [hunk 3]
```diff
@@ -65,13 +65,13 @@ end
 
 function show(io::IO, N::QuotientModule{T}) where T <: RingElement
    print(io, "Quotient module over ")
-   print(IOContext(io, :compact => true), base_ring(N))
+   print(terse(pretty(io)), Lowercase(), base_ring(N))
    show_gens_rels(io, N)
 end
 
 function show(io::IO, N::QuotientModule{T}) where T <: FieldElement
    println(io, "Quotient space over:")
```
[the remark this answers is not in the document]
Comment (fingolfin) [reply src/generic/QuotientModule.jl:73]: agreed

File: src/generic/FunctionField.jl (hunk 2 above)
Comment (fingolfin) [reply src/generic/FunctionField.jl]: fixed

**Comment by fingolfin** [comment]
The `singular letterplace Ring` vs `Singular letterplace Ring` is due to Singular.jl not using pretty printing at all so far. I've filed https://github.com/oscar-system/Singular.jl/issues/798 as a reminder.

Otherwise I hope this PR is now ready to go

**Review by lgoettgens — APPROVED** [review]

**Pull request merged**